smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/serde/member/MemberShapeDecodeGenerator.kt

```diff
@@ -168,6 +211,26 @@ open class MemberShapeDecodeGenerator(
                 is ListShape, is SetShape -> " ?? []"
                 // Maps can only have empty map as default value
                 is MapShape -> " ?? [:]"
+                is TimestampShape -> " ?? Date(timeIntervalSince1970: ${it.expectNumberNode().value})"
+                is DocumentShape -> {
+                    when {
+                        it.isObjectNode -> { " ?? Document.object([:])" }
+                        it.isArrayNode -> { " ?? Document.array([])" }
+                        it.isBooleanNode -> { " ?? Document.boolean(${it.expectBooleanNode().value})" }
+                        it.isStringNode -> { " ?? Document.string(\"${it.expectStringNode().value}\")" }
+                        it.isNumberNode -> { " ?? Document.number(${it.expectNumberNode().value})" }
+                        else -> {
+                            throw CodegenException("Document shape cannot have a default trait of ${it.type} type.")
+                        }
+                    }
+                }
+                is BlobShape -> {
+                    if (targetShape.hasTrait<StreamingTrait>()) {
+                        " ?? ByteStream.data(\"$it\".data(using: .utf8))"
+                    } else {
+                        " ?? \"$it\".data(using: .utf8)"
+                    }
+                }
```

Comment:
Add default value support to deserializer for timestamp, document, and blob shapes